### ISSUE TITLE
Don't render parameters which have default value?

### ISSUE THREAD

**Issue by sverhoeven** [i-VRESSE/workflow-builder#24]
For the global parameters if you filled the required fields (molecules, run_dir) then the toml has following parameters:

```yaml
concat = 1
mode = 'local'
ncores = 8
queue = ''
queue_limit = 100
self_contained = false
```

These all have the default values specified at https://github.com/haddocking/haddock3/blob/main/src/haddock/modules/defaults.yaml
Including them makes the file look very complicated, but also very explicit, so any changes to defaults will not change the reproducibility of the workflow.

Should the workflow config file exclude parameters which are set to the default?

**Comment by amjjbonvin** [comment]
Pretty much all parameters have default values… And you want to be able to change those.

What will be exposed should depend on the access level (easy, expert, …)
I.e. the form can hide some parameters depending on how you configure/define the level.

> 
> Should the workflow config file exclude parameters which are set to the default?
> 
>

**Comment by sverhoeven** [comment]
Agree, all the parameters with defaults that belong to the access level should be shown in the **form**. 

I am not sure if the generated workflow config **file** should have defaults in it.

**Comment by joaomcteixeira** [comment]
All your comments are true. The problem with the parameters described by @sverhoeven here is that `concat` and `queues` are only needed if `mode = hpc`. Parameters in `yaml` files have a `group` key. We can use it to define groups that show/hide altogether.

**Comment by sverhoeven** [comment]
Decided to when submitting a form to remove the parameters which are set to their default value. This will make the workflow configuration file only contain parameters which have been set by the user.
